source/crud/tests/unified/findOneAndReplace-let.yml

```diff
@@ -0,0 +1,133 @@
+description: "findOneAndReplace-let"
+
+schemaVersion: "1.4"
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents: [ commandStartedEvent ]
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name crud-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name coll0
+
+initialData: &initialData
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents:
+      - { _id: 1 }
+      - { _id: 2 }
+
+tests:
+  # TODO: Once SERVER-57403 is resolved, this test can be removed in favor of
+  # the "dollar-prefixed $literal value" test below.
```

Comment:
This seems like a good opportunity to update `aggregate-let.yml`. This should entail deleting one test and removing the topology restriction from the second test.

When doing so, I would suggest renaming "Aggregate with let option and dollar-prefixed $literal value" to "Aggregate with let option" since it will now be the _only_ non-error test (excluding the `$out` test towards the end of the file). "dollar-prefixed $literal value" was only added to the name to differentiate it from the first test, which we needed as a work-around for SERVER-57403. IIRC, there was really no reason to do `{ $literal: "bar" }` since `$literal` serves no purpose in that context. It's only necessary to prevent evaluation of a dollar-prefixed value.

To that end, I don't think there's any reason to mention "dollar-prefixed $literal value" in your other test titles. If you'd like your other tests to be comprehensive, you can consider using the same `let` bindings I used in the aggregation tests. There, I tried to cover all of the bases as mentioned in my test comment:

> This test demonstrates a basic constant value, a value wrapped with $literal (to avoid expression parsing), and a closed expression (e.g. $rand).

I also used an additional constant value to test variable usage with the `_id` field (`id` in my `let` bindings).

Reply:
I updated `aggregate-let.yml` as suggested. Further, I removed  "dollar-prefixed $literal value" from other tests in this PR. I agree there is no need to have it everywhere.